File: 3. Longest Substring Without Repeating Characters/LongestSubstringWithoutRepeatingCharacters.cs
public class Solution {
    public int LengthOfLongestSubstring(string s) {
        List<int> listOfCounts = new List<int>();

        string currentSubstring = "";

        foreach (char c in s) 
        {
            if (!currentSubstring.Contains(c))
            {
                currentSubstring += c;
            }
            else
            {
                listOfCounts.Add(currentSubstring.Length);
                int index = currentSubstring.IndexOf(c);
                currentSubstring = currentSubstring.Substring(index + 1) + c;
            }
        }
        listOfCounts.Add(currentSubstring.Length);
        int maxValue = listOfCounts.Count > 0 ? listOfCounts.Max() : 0;
        return maxValue;
        }
    }

File: 3. Longest Substring Without Repeating Characters/longest_substring_without_repeating_characters.py
class Solution(object):
    def lengthOfLongestSubstring(self, s):
        """
        :type s: str
        :rtype: int
        """
        list_of_counts = []
        current_substr = ""

        for char in s:
            if char not in current_substr:
                current_substr += char
            else:
                list_of_counts.append(len(current_substr))
                current_substr = current_substr[current_substr.index(char) + 1:] + char

        list_of_counts.append(len(current_substr))

        max_value = max(list_of_counts) if list_of_counts else 0
        return max_value